pkl-core/src/main/java/org/pkl/core/ast/type/TypeNode.java

```diff
@@ -2514,21 +2537,24 @@ public TypeNode initWriteSlotNode(int slot) {
     }
 
     @ExplodeLoop
-    public Object execute(VirtualFrame frame, Object value) {
-      if (customThisSlot == -1) {
-        CompilerDirectives.transferToInterpreterAndInvalidate();
-        // deferred until execution time s.t. nodes of inlined type aliases get the right frame slot
-        customThisSlot =
-            frame.getFrameDescriptor().findOrAddAuxiliarySlot(CustomThisScope.FRAME_SLOT_ID);
-      }
```

Comment:
As part of this fix, I also realized that this logic is incorrect. We don't know that this node will always be called from the same frame descriptor, because this same node might be called from different root nodes.

Added a test (constraints14.pkl) that covers this edge case.